unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2597,30 +2603,39 @@ protected void updateAllAgentCollidersForVisibilityCheck(bool enableColliders)
         }
 
         public SimObjPhysics[] VisibleSimObjs(string objectId, bool forceVisible = false) {
-            ServerAction action = new ServerAction();
-            action.objectId = objectId;
-            action.forceVisible = forceVisible;
-            return VisibleSimObjs(action);
+            SimObjPhysics sop = getSimObjectFromTypeOrId(objectType: null, objectId: objectId);
```

Comment:
This function's behavior is super strange to me given its name and inputs. Looking at how it's used in the code base I would vote for removing this completely and replacing it with something like:
```
public SimObjPhysics GetSimObjFromId(string objectId, bool excludeIfNotVisible)
```

Reply:
done, actually created two methods.  One that is just a method to get a SimObj by objectId and another that also does the interactable check with forceVisible boolean.